.github/workflows/deploy-ecs.yaml

```diff
@@ -0,0 +1,126 @@
+          jq --arg IMAGE_URI "$IMAGE_URI" --arg CONTAINER_NAME "$CONTAINER_NAME" \
+            '.containerDefinitions |= map(if .name == $CONTAINER_NAME then .image = $IMAGE_URI else . end) | del(.taskDefinitionArn, .revision, .status, .requiresAttributes, .placementConstraints, .compatibilities, .registeredAt, .registeredBy)' \
+            task-definition.json > updated-task-definition.json
+          
+          # Register the new task definition
+          NEW_TASK_DEFINITION_ARN=$(aws ecs register-task-definition \
+            --cli-input-json file://updated-task-definition.json \
+            --query 'taskDefinition.taskDefinitionArn' \
+            --output text)
+          
+          echo "New task definition ARN: $NEW_TASK_DEFINITION_ARN"
+          
+          # Update the ECS service with the new task definition
+          aws ecs update-service \
+            --cluster $ECS_CLUSTER \
+            --service $ECS_SERVICE \
+            --task-definition $NEW_TASK_DEFINITION_ARN \
+            --force-new-deployment
+          
+          # Wait for the service to be stable
+          echo "Waiting for ECS service to be stable..."
+          aws ecs wait services-stable \
+            --cluster $ECS_CLUSTER \
+            --services $ECS_SERVICE
+          
+          # Check the actual service status after waiting
+          echo "Checking service status..."
+          SERVICE_STATUS=$(aws ecs describe-services \
+            --cluster $ECS_CLUSTER \
+            --services $ECS_SERVICE \
+            --query 'services[0].status' \
+            --output text)
+          
+          echo "Service Status: $SERVICE_STATUS"
+          
+          # Check if deployment was successful
+          if [ "$SERVICE_STATUS" = "ACTIVE" ]; then
+            echo "Deployment completed successfully!"
+          else
+            echo "Deployment may have issues. Service status: $SERVICE_STATUS"
+            exit 1
+          fi
+          
+          # Output the deployment information
+          echo "deployed_image=$IMAGE_URI" >> $GITHUB_OUTPUT
+          echo "ecs_service=$ECS_SERVICE" >> $GITHUB_OUTPUT
+          echo "ecs_cluster=$ECS_CLUSTER" >> $GITHUB_OUTPUT
+          echo "task_definition_arn=$NEW_TASK_DEFINITION_ARN" >> $GITHUB_OUTPUT
+      - name: Get deployment status
+        run: |
+          echo "Deployment Status:"
+          aws ecs describe-services \
+            --cluster $ECS_CLUSTER \
+            --services $ECS_SERVICE \
+            --query 'services[0].{ServiceName:serviceName,Status:status,DesiredCount:desiredCount,RunningCount:runningCount,PendingCount:pendingCount,TaskDefinition:taskDefinition}' \
+            --output table
+          
+          echo ""
+          echo "Current running tasks:"
+          aws ecs list-tasks \
+            --cluster $ECS_CLUSTER \
+            --service-name $ECS_SERVICE \
+            --query 'taskArns' \
+            --output table
```

Comment:
check if we need to commit this file. zsa1 has it already.
same for all github action changes